storage/namespace.go

```diff
@@ -481,6 +481,9 @@ func (n *dbNamespace) NeedsFlush(blockStart time.Time) bool {
 		if state.Status == fileOpFailed && state.NumFailures < maxRetries {
 			return true
 		}
+		if state.Status == fileOpDirty {
```

Comment:
nit: could merge this with the condition check above into `if state.Status == fileOpNotStarted || state.Status == fileOpDirty`